packages/testing/src/execution_testing/cli/pytest_commands/plugins/shared/benchmarking.py

```diff
@@ -30,31 +32,68 @@ def pytest_addoption(parser: pytest.Parser) -> None:
     )
 
 
+def pytest_ignore_collect(
+    collection_path: Path, config: pytest.Config
+) -> bool | None:
+    """
+    Ignore tests/benchmark/ directory by default unless explicitly specified.
+
+    Returns True to ignore, False to collect, None for default behavior.
+    """
+    # Only handle paths within tests/benchmark/
+    try:
+        collection_path.relative_to(config.rootpath / "tests" / "benchmark")
+    except ValueError:
+        return None
+
+    # Check if benchmark tests explicitly specified in command line arguments
+    benchmark_path = config.rootpath / "tests" / "benchmark"
+    for arg in config.args:
+        arg_path = Path(arg)
+        # Check absolute paths
+        if arg_path.is_absolute():
+            try:
+                arg_path.relative_to(benchmark_path)
+                # Explicitly specified, set op_mode and don't ignore
+                config.op_mode = OpMode.BENCHMARKING  # type: ignore[attr-defined]
+                return False
+            except ValueError:
+                continue
+        # Check relative paths containing 'benchmark'
+        elif "benchmark" in arg:
```

Comment:
This is a rather loose check, perhaps we could check for "benchmark/compute" or "benchmark/stateful". Of course, it'll break if we add another valid benchmark sub-folder here.